### ISSUE TITLE
Non UTF-8 boot log crashes decoder

### ISSUE THREAD

**Issue by andschwa** [Azure/azure-cli#15590]
## Describe the bug

**Command Name**
`az vm boot-diagnostics get-boot-log`

**Errors:**
```
'utf-8' codec can't decode byte 0xff in position 20023: invalid start byte
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
...
storage/v2018_11_09/blob/baseblobservice.py, ln 2166, in get_blob_to_stream
    stream.write(blob.content)
cli/command_modules/vm/custom.py, ln 1438, in write
    content = str_or_bytes.decode('utf8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 20023: invalid start byte
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Deploy URN `audiocodes:mediantsessionbordercontroller:mediantvirtualsbcazure:latest`
- `az vm boot-diagnostics get-boot-log -n {} -g {}`

## Expected Behavior

Boot log to be downloaded.

## Environment Summary
```
Linux-4.19.128-microsoft-standard-x86_64-with-debian-bullseye-sid
Python 3.6.10
Installer: DEB

azure-cli 2.13.0

Extensions:
azure-devops 0.18.0

```

**Comment by yungezz** [comment]
hi @qwordy could  you pls help to look at this?

**Comment by qwordy** [comment]
The error indicates that it is not UTF-8 encoding. I will support more encodings to fix this issue.

**Comment by andschwa** [comment]
Thanks @qwordy! Believe me, I was quite surprised to find some VMs give a non-UTF-8 boot log, but so it goes.

**Comment by yungezz** [comment]
hi @qwordy is it ok to close the issue now? thanks

**Comment by qwordy** [comment]
> 
> 
> hi @qwordy is it ok to close the issue now? thanks

May be we can support more encodings.

**Comment by qwordy** [comment]
Mark it as a feature request: support more encodings of logs

**Comment by andschwa** [comment]
@qwordy I wouldn't categorize this as a feature request but as a bug fix, as this results in a crash when using this functionality of `az` against images available in the Azure marketplace.

**Comment by qwordy** [comment]
Yes. I also think it is a bug fix. The feature-request label has a long story. There is a similar issue https://github.com/Azure/azure-cli/issues/16454 recently. I will prioritize this issue since more than one person reported it.
